index.ts

```diff
@@ -176,30 +189,21 @@ wss.on("request", function (request) {
     };
   }
   const room: Room = rooms[roomId];
+  let conn: Connection = {
+    name,
+    connection,
+    id
+  };
+  room.connections.push(conn);
 
   connection.on("message", function (message) {
-    let name = "";
-    let id = "";
-
-    room.connections
-      .filter((item) => item.connection == connection)
-      .map((item) => {
-        name = item.name;
-        id = item.id;
-      });
-
     let data = JSON.parse(message.utf8Data);
     // console.log(`Inbound: ${message.utf8Data}`);
     let card: Card;
     let msg: string;
 
     switch (data.type) {
       case "join":
-        do {
-          id = generateId();
-        } while (room.connections.filter((item) => item.id == id).length > 0);
-
-        room.connections.push({ name: data.name, connection: connection, id: id });
         msg = JSON.stringify({
           type: "connected",
           connected: room.connections.map((item) => {
```

Comment:
One could remove the `join` type and move it to connection establishment as well. What do you think?

Reply:
What would be the advantage of that?
On the down side are the "costs" of changing this on server- and client-side. Also it makes extending the protocol harder when establishing connection is necessarily equal to `join`ing, see https://github.com/BDSU/meeting-moderation-cards/pull/8#discussion_r555373401